lib/plugins/cors.js

```diff
@@ -35,16 +35,25 @@ var AC_EXPOSE_HEADERS = 'Access-Control-Expose-Headers';
 
 ///--- Internal Functions
 
+// turn a string defining an allowed domain into a regex
+function toRegExp(str) {
+    if (str instanceof RegExp) {
+        return str;
+    }
+    var regexStr = str
+      // escape all the special characters, except *
+      .replace(/[\-\[\]\/\{\}\(\)\+\?\.\\\^\$\|]/g, '\\$&')
+      // replace all instances of * with a proper regex wildcard
+      .replace(/\*/g, '.*');
+
+    return new RegExp('^(https?://)?' + regexStr + '$', 'i');
+}
+
 function matchOrigin(req, origins) {
     var origin = req.headers.origin;
 
     function belongs(o) {
-        if (origin === o || o === '*') {
-            origin = o;
-            return (true);
-        }
-
-        return (false);
+        return o.test(origin);
```

Comment:
I'm not convinced we want to convert all normal strings origins into a RegExp. Instead, I think we should maintain existing behavior, but inside `matchOrigin()` special case the RegExp scenario vs string scenario. That would reduce the scope of changes for only RegExp values passed in.